dvc/stage/run.py

```diff
@@ -75,19 +74,11 @@ def get_executable():
 
 
 def _run(stage, executable, cmd, checkpoint_func, **kwargs):
-    main_thread = isinstance(
-        threading.current_thread(),
-        threading._MainThread,  # pylint: disable=protected-access
-    )
 
     exec_cmd = _make_cmd(executable, cmd)
-    old_handler = None
 
     try:
         p = subprocess.Popen(exec_cmd, **kwargs)
-        if main_thread:
```

Comment:
@efiop ,

Excuse me, these `signal.signal(signal.SIGINT, signal.SIG_IGN)` blocks all the interrupts sent to the running stage. This signal is useful when we were doing some checking point training and want an early stopping. These codes were introduced 3 years ago, I don't have the context of why they are added here, could you please help me to find a solution here?

Reply:
I think the issue is that this is not storing the old signal handler. We want to ignore sigint in the parent process (in `main_thread`) after calling `subprocess.Popen`, and then we want to un-ignore it after the child finishes running. However, the current code just makes us ignore the signal in the parent without saving the original handler (so it never gets un-ignored later).

It should probably be
```
if main_thread:
    old_handler = signal.getsignal(signal.SIGINT)
    signal.signal(signal.SIGINT, signal.SIG_IGN)
```